hw02/Homwork 2.Rmd

```diff
@@ -0,0 +1,137 @@
+---
+title: "Homework 2"
+author: "Samruddhi Shinde"
+date: "2/7/2021"
+output:
+  html_document: default
+  github_document: default
+---
+#Exercise 1
+
+##1.1 
+
+Loading in packages needed for assignments
+```{r}
+library(tidyverse)
+library(gapminder)
+```
+
+Filter data set to show 3 countries (Mexico, Pakistan, & Zimbabwe) from the 1970s
+```{r}
+gapminder %>%
+  filter(country %in% c("Mexico","Pakistan", "Zimbabwe") & year %in% c(1970:1979))
+```
+
+##1.2
+
+Select country and gdpPercap from filtered dataset
+```{r}
+gapminder %>%
+  filter(country %in% c("Mexico","Pakistan", "Zimbabwe") & year %in% c(1970:1979)) %>%
+  select(country, gdpPercap)
+```
+
+##1.3 
+
+Create new variable for thr change in life expectancy
+```{r}
+gapminder %>%
+  mutate(change_in_lifeExp = lifeExp - lag(lifeExp)) %>%
+  filter(change_in_lifeExp<0)
+```
+
+##1.4
+
+Filter gapminder dataset to show the max GDP per capita for each country
+```{r}
+gapminder %>%
+  group_by(country) %>%
+  summarize(max(gdpPercap))
+```
+
+##1.5
+
+Scatter plot of Canada's Life expectancy vs GDP per capita
+```{r}
+gapminder %>%
+  filter(country == "Canada") %>%
+  ggplot +
+    aes(x = gdpPercap, y = lifeExp) +
+    geom_point() + 
+    scale_x_log10("GDP per capita", labels = scales::dollar_format()) +
+    ylab("Life Expectancy") +
+    ggtitle("Canada's Life Expectancy vs GDP per Capita")
+```
+
+#Excerise 2
+
+##2.1
+
+Loading in packages for exercise 
+
+```{r}
+library(palmerpenguins)
+library(tidyverse)
+```
+
+Summarize bill length and bill depth
+```{r}
+palmerpenguins::penguins %>%
+  select(bill_length_mm, bill_depth_mm) %>%
+  summarize(
+    meanLength = mean(bill_length_mm, na.rm = TRUE),
+    sdLength = sd(bill_length_mm, na.rm = TRUE),
+    meanDepth = mean(bill_depth_mm, na.rm = TRUE),
+    sdDepth = sd(bill_depth_mm, na.rm = TRUE))
+```
+
+Scatter plot comparing bill length and bill depth
+```{r}
+palmerpenguins::penguins %>%
+  ggplot +
+    aes(x = bill_length_mm, y = bill_depth_mm) +
+    geom_point(alpha = 0.5) +
+    scale_x_log10("Bill Length") +
+    ylab("Bill Depth")
+```
+
+##2.2
+
+Sample size of population
+```{r}
+palmerpenguins::penguins %>%
+  select(species, body_mass_g) %>%
+    summarize(rows = n())
+```
+
+Body mass categorized by species
+```{r}
+palmerpenguins::penguins %>%
+  select(species, body_mass_g) %>%
+  group_by(species) %>%
+    summarize(meanMass = mean(body_mass_g, na.rm = TRUE))
```

Comment:
Instead of creating two summary tables, I think the table is supposed to have the sample size and a summary of the selected quantitative variable. Adding 'n()' as an argument within the summary function will create a column with the sample size.